embedded/store/tx.go

```diff
@@ -272,68 +304,78 @@ func (tx *Tx) Proof(key []byte) (*htree.InclusionProof, error) {
 }
 
 func (tx *Tx) readFrom(r *appendable.Reader) error {
+	tx.header = &TxHeader{}
+
 	id, err := r.ReadUint64()
 	if err != nil {
 		return err
 	}
-	tx.ID = id
+	tx.header.ID = id
 
 	ts, err := r.ReadUint64()
 	if err != nil {
 		return err
 	}
-	tx.Ts = int64(ts)
+	tx.header.Ts = int64(ts)
 
 	blTxID, err := r.ReadUint64()
 	if err != nil {
 		return err
 	}
-	tx.BlTxID = blTxID
+	tx.header.BlTxID = blTxID
 
-	_, err = r.Read(tx.BlRoot[:])
+	_, err = r.Read(tx.header.BlRoot[:])
 	if err != nil {
 		return err
 	}
 
-	_, err = r.Read(tx.PrevAlh[:])
+	_, err = r.Read(tx.header.PrevAlh[:])
 	if err != nil {
 		return err
 	}
 
-	mdLen, err := r.ReadUint16()
+	version, err := r.ReadUint16()
 	if err != nil {
 		return err
 	}
+	tx.header.Version = int(version)
 
-	var txmd *TxMetadata
+	if tx.header.Version > 0 {
```

Comment:
Shouldn't we explicitly check for the version value here? What if there's a header with version 2 (e.g. after some future update of immudb)? We should not continue in such case.

Reply:
will take a second round on this, so to make the version more explicit the version